runtime/drivers/duckdb/olap.go

```diff
@@ -257,11 +270,23 @@ func (c *connection) InsertTableAsSelect(ctx context.Context, name, sql string,
 
 	if strategy == drivers.IncrementalStrategyMerge {
 		err = db.MutateTable(ctx, name, func(ctx context.Context, conn *sqlx.Conn) error {
+			// Execute the pre-init SQL first
+			if beforeInsert != "" {
+				_, err := conn.ExecContext(ctx, beforeInsert)
+				return err
+			}
+			aferExec := func() error {
+				if afterInsert != "" {
+					_, err := conn.ExecContext(ctx, afterInsert)
+					return err
+				}
+				return nil
+			}
 			// Create a temporary table with the new data
 			tmp := uuid.New().String()
 			_, err := conn.ExecContext(ctx, fmt.Sprintf("CREATE TEMPORARY TABLE %s AS (%s\n)", safeSQLName(tmp), sql))
 			if err != nil {
-				return err
+				return errors.Join(err, aferExec())
```

Comment:
Since this is duplicated four times (easy to mess up for the next maintainer), it would be better to use `defer` and overwrite the function return values if it errors.